test/utils.cpp

```diff
@@ -119,6 +119,17 @@ TEST_CASE("Tokenizers behave correctly", "[utils]") {
 	}
 }
 
+TEST_CASE("utils::consolidate_whitespace replaces multiple consecutinve whitespace with a single space", "[utils]") {
+	REQUIRE(utils::consolidate_whitespace("LoremIpsum") == "LoremIpsum");
+	REQUIRE(utils::consolidate_whitespace("Lorem Ipsum") == "Lorem Ipsum");
+	REQUIRE(utils::consolidate_whitespace("    Lorem \t\tIpsum \t ") == " Lorem Ipsum ");
+	REQUIRE(utils::consolidate_whitespace("    Lorem \r\n\r\n\tIpsum") == " Lorem Ipsum");
+
+	REQUIRE(utils::consolidate_whitespace("") == " ");
```

Comment:
Oh, I didn't consider that! I'd argue that this behaviour is confusing because function's name promises to remove, not add, spaces. Do we rely on it?